src/types.md

```diff
@@ -428,13 +499,60 @@ Box<Bar<'a>>
 Box<Bar<'a> + 'a>
 ```
 
-The default for user-defined trait objects is based on the object type itself.
-If a type parameter has a lifetime bound, then that lifetime bound becomes the
-default bound for trait objects of that type. For example, `std::cell::Ref<'a,
-T>` contains a `T: 'a` bound, therefore trait objects of type `Ref<'a,
-SomeTrait>` are the same as `Ref<'a, (SomeTrait + 'a)>`.
-
-### Type parameters
+The default for user-defined types with a trait object used as a type parameter
+is based on the type itself. If a type parameter has a lifetime bound, then
+that lifetime bound becomes the default bound for trait objects of that type.
+For example, `std::cell::Ref<'a, T>` contains a `T: 'a` bound, therefore trait
+objects of type `Ref<'a, SomeTrait>` are the same as `Ref<'a, (SomeTrait +
+'a)>`. If there's no such bound then the default bound is `'static`, for
+example in `std::rc::Rc<T>`.
+
+
+## Dynamically sized types
+
+Most types have a fixed size that is known at compile time and implement the
+trait [`Sized`]. A type with a size that is known only at run-time is called a
+_dynamically sized type_ (_DST_) or unsized type. [Slices] and [trait objects]
+are two of the most DSTs. Such types can only be used in certain cases:
+
+* [Pointer types] to DSTs are sized but have twice the size of pointers to
+  sized types
+    * Pointers to slices also store the number of elements of the slice.
+    * Pointers to trait objects also store a pointer to a vtable.
+* Traits may be implemented for DSTs, this means that `Self` is not assumed to
```

Comment:
Should definitely be a period instead of a comma after DSTs. I think this can be shortened considerably. Consider the following, where `elsewhere` is a link to wherever talks about the Sized trait and where it is automatically inferred.

"Traits can be implemented for DSTs. Unlike elsewhere `Self: ?Sized` by default in trait definitions."

Reply:
Link to https://doc.rust-lang.org/std/marker/trait.Sized.html?

Reply:
How about https://doc.rust-lang.org/nightly/reference/the-sized-trait.html and a later PR, we can fill in that section?